### PR TITLE
Update chokidar

### PR DESCRIPTION
**Pull request by greggman**
Note: This is dependent on https://github.com/webgpu/webgpu-samples/pull/463

Update chokidar to v4

chokidar v3 is no longer supported. v4 requires node 20 so

1. Added a check for v20+

2. Make script exit if any spawn gets an error.

   This required watching for processes to exit and
   killing the other processes.

3. Update to use glob as chokidar v4 no longer does it

Also fixed a minor warning in sample/utis.ts

### CODE REVIEW

File: sample/util.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ export function quitIfWebGPUNotAvailable(
   if (!device) {
     quitIfAdapterNotAvailable(adapter);
     fail('Unable to get a device for an unknown reason');
+    return;
```
Comment (kainino0x) [inline sample/util.ts:25]: IIRC `fail` should always throw, so this should be dead code?

File: build/tools/serve.js [hunk 2]
```diff
@@ -3,17 +3,29 @@ import { mkdirSync } from 'fs';
 
 mkdirSync('out', { recursive: true });
 
-spawn('npm', ['run', 'watch'], {
+const spawns = [];
+
+function spawnAndCheck(cmd, args, options) {
+  const s = spawn(cmd, args, options);
+  spawns.push(s);
```
Comment (kainino0x) [inline build/tools/serve.js:10]: doesn't seem like items are ever removed from this array. If that's not a problem, add a comment about why.

File: sample/util.ts (hunk 1 above)
Comment (greggman) [reply sample/util.ts:25]: Without this I get these errors from the lines below.

```
(!) [plugin typescript] sample/util.ts (27:3): @rollup/plugin-typescript TS18047: 'device' is possibly 'null'.
/Users/gman/src/webgpu-samples/sample/util.ts:27:3

27   device.lost.then((reason) => {
     ~~~~~~
```

File: build/tools/serve.js (hunk 2 above)
Comment (greggman) [reply build/tools/serve.js:10]: done

File: sample/util.ts (hunk 1 above)
Comment (kainino0x) [reply sample/util.ts:25]: ahh, that makes sense, this is probably too complex of a scenario to trigger type narrowing on `device` even though `fail` presumably has a return type of `never`.

File: build/tools/serve.js [hunk 3]
```diff
@@ -9,9 +9,12 @@ function spawnAndCheck(cmd, args, options) {
   const s = spawn(cmd, args, options);
   spawns.push(s);
   s.on('close', (code) => {
-    console.log(cmd, 'exited with code:', code);
-    spawns.forEach((s) => s.kill());
-    process.exit(code);
+    spawns.splice(spawns.indexOf(s), 1);
```
Comment (kainino0x) [inline build/tools/serve.js:12]: nit: maybe `spawns` could be a `Set`? but not important

**Review by kainino0x — APPROVED** [review]

**Pull request merged**